### PR TITLE
[DIS] new `resource/opentelekomcloud_dis_stream_v2`

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
## Summary of the Pull Request
New resource

## PR Checklist

* [x] Refers to: #2198
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccDisStreamV2_basic
=== PAUSE TestAccDisStreamV2_basic
=== CONT  TestAccDisStreamV2_basic
--- PASS: TestAccDisStreamV2_basic (80.54s)
PASS


Process finished with the exit code 0
```

### CODE REVIEW

**Review by Aloento — DISMISSED** [review]

File: opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+				Required: true,
+			},
+			"retention_period": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      24,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(24, 72),
+			},
+			"stream_type": {
```
Comment (artem-lifshits) [inline opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go:52]: Please include `ValidateFunc` here

File: opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"data_type": {
```
Comment (artem-lifshits) [inline opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go:61]: Same here

File: opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+			"compression_format": {
```
Comment (artem-lifshits) [inline opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go:70]: Same here

File: opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+		Tags:              common.ExpandResourceTags(d.Get("tags").(map[string]interface{})),
+	}
+
+	autoScaleMinPartitionCount := d.Get("auto_scale_min_partition_count").(int)
+	autoScaleMaxPartitionCount := d.Get("auto_scale_max_partition_count").(int)
+	if autoScaleMinPartitionCount > 0 && autoScaleMaxPartitionCount > 0 {
+		opts.AutoScaleEnabled = pointerto.Bool(true)
+		opts.AutoScaleMinPartitionCount = &autoScaleMinPartitionCount
+		opts.AutoScaleMaxPartitionCount = &autoScaleMaxPartitionCount
+	} else {
```
Comment (artem-lifshits) [inline opentelekomcloud/services/dis/resource_opentelekomcloud_dis_stream_v2.go]: redundant else

**Review by artem-lifshits — APPROVED** [review]

**Review by Aloento — APPROVED** [review]

**Pull request merged**